### PR TITLE
[quant][graphmode][fx] Change standalone module api

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49754 [quant][graphmode][fx] Standalone module support {input/output}_quantized_idxs
* **#49719 [quant][graphmode][fx] Change standalone module api**

Summary:
We find there are multiple use cases for standalone module, one use case requires standalone module
to produce a module that takes float Tensor as input and outputs a float Tensor, the other needs to
produce a modulee that takes quantized Tensor as input and outputs a quantized Tensor.

This is similar to `quantized_input_idxs` and `quantized_output_idxs` so we want to nest
prepare_custom_config_dict in the standalone module configuration, for maximum flxibility we also
include qconfig_dict for stand alone module as well in case user needs to have special qconfig_dict for
the standalone module in the future.

Changed from
```python
prepare_custom_config_dict =
{
  "standalone_module_name": ["standalone_module"],
   "standalone_module_class": [StandaloneModule]
 }
```
to
```python
prepare_custom_config_dict =
{
  "standalone_module_name": [("standalone_module", qconfig_dict1, prepare_custom_config_dict1)],
  "standalone_module_class": [(StandaloneModule, qconfig_dict2, prepare_custom_config_dict2)]
 }
```
The entries in the config are:
1. name/module_class
2. optional qconfig_dict, when it is None, we'll use {"": qconfig} where qconfig is the one from parent qconfig_dict
3. optional prepare_custom_config_dict, when it is None, we'll use default value of prepare_custom_config_dict for prepare API (None)

Test Plan:
python test/test_quantization.py TestQuantizeFx.test_standalone_module

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25675704](https://our.internmc.facebook.com/intern/diff/D25675704)

### CODE REVIEW

File: torch/quantization/quantize_fx.py [hunk 1]
```diff
@@ -178,11 +178,13 @@ def prepare_fx(
         # optional: specify the path for standalone modules
         # These modules are symbolically traced and quantized as one unit
         "standalone_module_name": [
-           "submodule.standalone"
+           # module_name, qconfig_dict, prepare_custom_config_dict
+           ("submodule.standalone", None, None)
```
Comment (vkuzo) [inline torch/quantization/quantize_fx.py:182]: does `qconfig_dict` == None here means "do not quantize" or "use parent qconfig"?

Comment (jerryzh168) [reply torch/quantization/quantize_fx.py:182]: it means "use parent qconfig"
for "do not quantize": qconfit_dict = {"": None}

File: test/quantization/test_quantize_fx.py [hunk 2]
```diff
@@ -575,8 +575,8 @@ def forward(self, x):
         original_ref_m.conv2.bias = torch.nn.Parameter(original_m.standalone.conv.bias.detach())
 
         qconfig_dict = {"": default_qconfig}
-        config_name = {"standalone_module_name": ["standalone"]}
-        config_class = {"standalone_module_class": [StandaloneModule]}
+        config_name = {"standalone_module_name": [("standalone", None, None)]}
```
Comment (raghuramank100) [inline test/quantization/test_quantize_fx.py:578]: Can we also test for selective quantization of inputs/outputs for standalone modules?

Comment (jerryzh168) [reply test/quantization/test_quantize_fx.py:578]: yes, it's in the next PR

Comment (raghuramank100) [reply test/quantization/test_quantize_fx.py:578]: Sorry, just saw that

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**